ApplicationLibCode/Commands/ExportCommands/RicSnapshotAllViewsToFileFeature.cpp

```diff
@@ -86,59 +85,68 @@ void RicSnapshotAllViewsToFileFeature::exportSnapshotOfViewsIntoFolder( const QS
-    for ( size_t i = 0; i < projectCases.size(); i++ )
+    if ( ( caseId == -1 ) && ( viewId == -1 ) )
     {
-        RimCase* cas = projectCases[i];
-        if ( !cas ) continue;
-
-        bool matchingCaseId = caseId == -1 || caseId == cas->caseId();
-        if ( !matchingCaseId ) continue;
+        viewsForSnapshot = project->allViews();
+    }
+    else
+    {
+        std::vector<RimCase*> projectCases = project->allGridCases();
+        for ( size_t i = 0; i < projectCases.size(); i++ )
+        {
+            RimCase* cas = projectCases[i];
+            if ( !cas ) continue;
 
-        std::vector<Rim3dView*> views = cas->views();
+            bool matchingCaseId = caseId == -1 || caseId == cas->caseId();
+            if ( !matchingCaseId ) continue;
 
-        for ( size_t j = 0; j < views.size(); j++ )
-        {
-            Rim3dView* riv = views[j];
+            std::vector<Rim3dView*> views = cas->views();
 
-            if ( riv && riv->viewer() && ( viewId == -1 || viewId == riv->id() ) )
+            for ( size_t j = 0; j < views.size(); j++ )
             {
-                RiaApplication::instance()->setActiveReservoirView( riv );
+                Rim3dView* riv = views[j];
 
-                RiuViewer* viewer = riv->viewer();
-                Riu3DMainWindowTools::setActiveViewer( viewer->layoutWidget() );
+                if ( riv && riv->viewer() && ( viewId == -1 || viewId == riv->id() ) )
+                {
+                    viewsForSnapshot.push_back( riv );
+                }
+            }
+        }
+    }
 
-                RiaViewRedrawScheduler::instance()->clearViewsScheduledForUpdate();
-                RiaPlotWindowRedrawScheduler::instance()->clearAllScheduledUpdates();
+    const QString absSnapshotPath = snapshotPath.absolutePath();
+    RiaLogging::info( QString( "Exporting snapshot of all views to %1" ).arg( snapshotFolderName ) );
 
-                // riv->updateCurrentTimeStepAndRedraw();
-                riv->createDisplayModelAndRedraw();
-                viewer->repaint();
+    for ( auto riv : viewsForSnapshot )
+    {
+        RiaApplication::instance()->setActiveReservoirView( riv );
 
-                QString fileName = RicSnapshotFilenameGenerator::generateSnapshotFileName( riv );
-                if ( !prefix.isEmpty() )
-                {
-                    fileName = prefix + fileName;
-                }
+        RiuViewer* viewer = riv->viewer();
+        Riu3DMainWindowTools::setActiveViewer( viewer->layoutWidget() );
 
-                QString absoluteFileName = caf::Utils::constructFullFileName( absSnapshotPath, fileName, ".png" );
+        RiaViewRedrawScheduler::instance()->clearViewsScheduledForUpdate();
+        RiaPlotWindowRedrawScheduler::instance()->clearAllScheduledUpdates();
 
-                RicSnapshotViewToFileFeature::saveSnapshotAs( absoluteFileName, riv );
+        riv->createDisplayModelAndRedraw();
+        viewer->repaint();
```

Comment:
Should probably call QWidget::update instead?
https://doc.qt.io/qt-6/qwidget.html#repaint

Reply:
We need immediate repaint here, as the pixels are read out after this function.